### PR TITLE
planner: fix a panic caused by sinking a Limit with inlined Proj into IndexLookUp when accessing a partition table

### LINKED ISSUE #24636: Wrong value when query a partition table with `LIMIT`
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
Scripts:
```
drop table if exists test_partition;

CREATE TABLE `test_partition` (
  `a` varchar(100) NOT NULL,
  `b` date NOT NULL,
  `c` varchar(100) NOT NULL,
  `d` datetime DEFAULT NULL,
  `e` datetime DEFAULT NULL,
  `f` bigint(20) DEFAULT NULL,
  `g` bigint(20) DEFAULT NULL,
  `h` bigint(20) DEFAULT NULL,
  `i` bigint(20) DEFAULT NULL,
  `j` bigint(20) DEFAULT NULL,
  `k` bigint(20) DEFAULT NULL,
  `l` bigint(20) DEFAULT NULL,
  PRIMARY KEY (`a`,`b`,`c`) /*T![clustered_index] NONCLUSTERED */
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin
PARTITION BY RANGE ( TO_DAYS(`b`) ) (
  PARTITION `pmin` VALUES LESS THAN (737821),
  PARTITION `p20200601` VALUES LESS THAN (738289)
);

INSERT INTO test_partition (a, b, c, d, e, f, g, h, i, j, k, l) VALUES('aaa', '2021-05-05', '428ff6a1-bb37-42ac-9883-33d7a29961e6', '2021-05-06 08:13:38', '2021-05-06 13:28:08', 0, 8, 3, 0, 9, 1, 0);

select c,j,l
from test_partition 
where `c`='428ff6a1-bb37-42ac-9883-33d7a29961e6' and `a`='aaa';

select c,j,l
from test_partition 
where `c`='428ff6a1-bb37-42ac-9883-33d7a29961e6' and `a`='aaa' limit 0, 200;
```

### 2. What did you expect to see? (Required)
```
tidb> select c,j,l from test_partition  where `c`='428ff6a1-bb37-42ac-9883-33d7a29961e6' and `a`='aaa';
+--------------------------------------+------+------+
| c                                    | j    | l    |
+--------------------------------------+------+------+
| 428ff6a1-bb37-42ac-9883-33d7a29961e6 |    9 |    0 |
+--------------------------------------+------+------+
1 row in set (0.00 sec)

tidb> select c,j,l from test_partition  where `c`='428ff6a1-bb37-42ac-9883-33d7a29961e6' and `a`='aaa' limit 0, 200;
+--------------------------------------+------+------+
| c                                    | j    | l    |
+--------------------------------------+------+------+
| 428ff6a1-bb37-42ac-9883-33d7a29961e6 |    9 |    0 |
+--------------------------------------+------+------+
1 row in set (0.00 sec)
```

### 3. What did you see instead (Required)
```
tidb> select c,j,l from test_partition  where `c`='428ff6a1-bb37-42ac-9883-33d7a29961e6' and `a`='aaa';
+--------------------------------------+------+------+
| c                                    | j    | l    |
+--------------------------------------+------+------+
| 428ff6a1-bb37-42ac-9883-33d7a29961e6 |    9 |    0 |
+--------------------------------------+------+------+
1 row in set (0.00 sec)

tidb> select c,j,l from test_partition  where `c`='428ff6a1-bb37-42ac-9883-33d7a29961e6' and `a`='aaa' limit 0, 200;
+-----+---------------------+------+
| c   | j                   | l    |
+-----+---------------------+------+
| aaa | 3558184994028925492 |    9 |
+-----+---------------------+------+
1 row in set (0.00 sec)
```

### 4. What is your TiDB version? (Required)

TiDB v5.0.1

### Other versions 
It is not reproduced in v4.0, but in current master(aecff1c42e3a05b911311f695db66fce228db369), an error is reported:

```
tidb> select c,j,l from test_partition  where `c`='428ff6a1-bb37-42ac-9883-33d7a29961e6' and `a`='aaa' limit 0, 200;
ERROR 1105 (HY000): Internal error: UnionExec chunk column count mismatch, req: 3, result: 4
```
So I'll assume it is not a known bug.

Update: when the table is specified as `clustered, the error doesn't reproduce.

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve?

Issue Number: close #24636 

Problem Summary: planner: fix a panic caused by sinking a Limit with inlined Proj into IndexLookUp when accessing a partition table

### What is changed and how it works?
The root cause is that the two features Limit Sinking and Inline Projection are not compatible with each other under partition tables.

### Check List 

Tests 

- Unit test

### Release note 

- planner: fix a panic caused by sinking a Limit with inlined Proj into IndexLookUp when accessing a partition table

### CODE REVIEW

File: planner/core/task.go [hunk 1]
```diff
@@ -1121,6 +1121,11 @@ func (p *PhysicalLimit) sinkIntoIndexLookUp(t task) bool {
 			return false
 		}
 	}
+	// If this happens, some Projection Operator must be inlined into this Limit.
+	// For safety, do not sink Limit into this LookUp in this case.
+	if p.Schema().Len() != reader.Schema().Len() {
```
Comment (winoros) [inline planner/core/task.go:1126]: Can we post a plan tree as an example?

Comment (qw4990) [reply planner/core/task.go:1130]: More comments are added, PTAL @winoros

**Review by rebelice — APPROVED** [review]

**Review by winoros — APPROVED** [review]

**Pull request merged**